src/mechanics/IGA/NURBS.h

```diff
@@ -0,0 +1,478 @@
+#pragma once
+
+#include<eigen3/Eigen/Dense>
+#include <array>
+#include <vector>
+#include <iostream>
+#include "mechanics/nodes/NodeSimple.h"
+
+namespace NuTo
+{
+template <int TDimParameter>
+class NURBS
+{
+public:
+
+    enum mParametrization {chord, centripetal};
+
+    /** Constructors **/
+
+    //! @brief ... default constructor
+    NURBS(){}
+
+    //! @brief ... constructor
+    //! @param rDegree ... degree of the polynomial
+    //! @param rKnots ... knot vector
+    //! @param rControlPoints ... control points
+    NURBS(const std::array<std::vector<double>, TDimParameter> &rKnots,
+          const std::vector<std::vector<NodeSimple>>           &rControlPoints,
+          const std::vector<std::vector<double>>               &rWeights,
+          const std::array<int,TDimParameter>                  &rDegree)
+        : mKnots(rKnots), mControlPoints(rControlPoints), mWeights(rWeights), mDegree(rDegree)
+    {}
+
+    /** Getter **/
+
+    //! @brief ... get the dimension of the curve
+    //! @return ... dimension of the curve
+    int GetDimension() const
+    {
+        return mControlPoints[0][0].GetNumValues();
+    }
+
+    int GetNumControlPointsElement(int dir) const
+    {
+        assert(dir <= TDimParameter && dir >= 0);
+        return mDegree[dir]+1;
+    }
+
+    int GetNumControlPointsElement() const
+    {
+        int numCPs = 1;
+        for(int i = 0; i < TDimParameter; i++)
+            numCPs *= mDegree[i]+1;
```

Comment:
Consider a range-based for loop.
~~~cpp
for(int degree : mDegree)
    numCPs *= degree + 1;
~~~